### ISSUE TITLE
HttpClient.get method does not throw an error on passing null for URL

### ISSUE THREAD

**Issue by kajalsinha12** [angular/angular#33879]
The HttpClient.get method from @angular/common/http does not throw an error on passing null for the URL parameter. This should be handled in the framework itself, otherwise the developer has to  put a check for handling the same.

**Comment by dannyskoog** [comment]
I would argue that there’s a major malfunction on the framework consumer side if `null` were to ever be passed as `url` to the `HttpClient.get` function. Especially since passing a proper `string` for that particular parameter is fundamental for the function itself (which is obvious).

Therefore the validation in this case should happen on the consumer side IMHO.

**Comment by kajalsinha12** [comment]
I agree to the part where you say that the consumer side should not pass null, but nobody would intentionally do that. Few examples I can give where unintentionally null or undefined could be passed are as below:
1. I had by mistake declared my class variable as `private url: 'api/products/products.json'`. And I was using the `url` variable in the `HttpClient.get` method. The application did not throw any exception at all.
2. Let say the `HttpClient.get` method is enclosed in a method X with some other piece of code. The method X has the `url` as a parameter which unknowingly is being passed as `null or undefined` by the caller.

I feel in such scenarios instead the developer doing the checks before every `HttpClient.get` method, the framework should throw an exception. 

Happy to hear your opinion.

**Comment by dannyskoog** [comment]
Thanks for those examples!

I mean, I believe you should always create an abstraction layer between your own codebase and certain 3rd party functionalities. And in this case preferably some sort of `HttpService` that wraps the functionality of `HttpClient`.

By doing this it would be fairly easy to add a guard just before your single usage of `HttpClient.get`

**Comment by ajitsinghkaler** [comment]
@dannyskoog I think what @kajalsinha12 is asking shouldn't angular throw an error in these cases by default instead of a wrapper implementation of it.

**Comment by dannyskoog** [comment]
Yes, I know that @ajitsinghkaler. I was just pointing out that IMHO it shouldn’t matter if Angular throws an error or not in this particular case. Since you should already be validating your own data, preferably in an abstraction layer (= good practice)

**Comment by kajalsinha12** [comment]
Agreed @dannyskoog to your point that validating data should be taken care by the developer as it is a good practice. The only point I am trying to make is the framework throwing exception for null and undefined would also be a good practice IMHO, thus should be included.

**Comment by dannyskoog** [comment]
And I respect your opinion @kajalsinha12. Let’s see what others have to say in this matter
